tests/index/names.c

```diff
@@ -25,6 +25,42 @@ void test_index_names__cleanup(void)
 	cl_git_sandbox_cleanup();
 }
 
+static void index_add_main(void)
```

Comment:
In the commit message you talk of "This test", but this commit doesn't do anything but add a new function which is never being called. The actual caller is only being added in the last commit of this series, which is a confusing when reading commits one by one

Reply:
By the way, one more thing: I don't really get what "main" is referring to in this context.

Reply:
Thanks; this got jumbled while rebasing.  I fixed it, and I also changed the name to `add_conflicts`.  Regarding "main", I was referring to the "main index" (as opposed to the other sections like the tree cache) but I updated it to avoid confusion.